transport/client.go

```diff
@@ -0,0 +1,135 @@
+package transport
```

Comment:
Should this client be in a different package ? `transport` sounds like a weird choice.

Reply:
Kinda feels like it maybe should, but it also is a universal client for various transports. Though if we are getting away from the transport name and going with streamconn  it might make sense to move it.